inausoft.netCLI.all/inausoft.netCLI/RootCommandHandler.cs

```diff
@@ -52,7 +52,7 @@ public int Run(string[] args)
                 {
                     throw new InvalidCommandException(commandName, $"Command {commandName} is invalid.");
                 }
-
+                //TODO: I think that this part is overcomplicated. See MateuszCommand Folder
                 return commandHandler.Run(CreateCommandFromExpression(commandHandler.GetCommandType(), fullExpresion));
```

Comment:
I think all .xml or .json parser work in exact similar way - but are probably optimized in terms of performance. As in most cases we would need to have this parsed only once for relativelly short expression I don't the point for optimalization.
This part is actually something that I would abstract from at some later point to something like:
```
public interface ICommandLineParser
{
    T CreateFromString<T>(string fullExpresion)
}
```

and let user implement it accordingly.

Reply:
I would avoid using reflection whenever it is possible - it adds unnecessary complexity to the solution - that was purpose of my comment. 
I have some troubles understanding how it works and I'm pretty sure that this part could be make easier (but i don't know how :P )

Reply:
Take a look at https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Utilities/ReflectionUtils.cs when you will study popular .json converter nuget it is full of `Reflection` usage. If you will read a bit I hope you will came to conclusion that deserializing is one of the most justified use of reflection - and it is just the case when all RootCommandHandler do is actually:
deserialize 'move --pathTo C:\samplefolder\myfolder' into object.

Reply:
I got the point that reflection is the only way that you are able to get infromation from annotations. Agreed to proposed implementation